marconi-core/src/Marconi/Core/Indexer/SQLiteAggregateQuery.hs

```diff
@@ -83,7 +83,7 @@ mkSQLiteAggregateQuery
 mkSQLiteAggregateQuery sources = do
   con <- liftIO $ SQL.open ":memory:"
   let databaseFromSource :: SQLiteSourceProvider m point -> IO FilePath
-      databaseFromSource (SQLiteSourceProvider ix) = Con.withMVar ix (pure . getDatabasePath)
+      databaseFromSource (SQLiteSourceProvider ix) = Con.withMVar ix (pure . SQLite.extractStorageUnsafe . getDatabasePath)
```

Comment:
Remove `extractStorageUnsafe`, and explicitly throw an `Exception` if one of the source is an in-memory db.